### ISSUE TITLE
How do I update development packages?

### ISSUE THREAD

**Issue by protist** [polygamma/aurman#253]
In the past, to update the out-of-date development versions of packages, I'd just run
```
$ aurman -Syu --devel --needed --noedit --noconfirm
```

However, in `aurman 2.18-1` this now produces an error message:
```
$ aurman -Syu --devel --needed --noedit --noconfirm
...
~~ initializing aurman...
!! There are unfulfilled dependencies of development packages.
   Please fulfill them first
   or use --devel_skip_deps
   or use --optimistic_versioning or --ignore_versioning
   or do not use --devel
   or use --ignore to ignore respective packages
:: aur/neomutt-git misses lynx
:: aur/nodejs-imapnotify-git misses npm
:: aur/plasma5-applets-resources-monitor-git misses extra-cmake-modules
:: aur/xsettingsd-git misses scons
:: aur/xss-lock-git misses python-docutils
```

I checked the named packages, and these are all makedepends that I do not have installed. I read through the aurman help, but I'm a little confused. `--devel_skip_deps` says `Skips dependency checks when determining development packages versions`, so I thought it would skip checking dependencies until *after* the latest package development versions were confirmed, but this didn't appear to check for new versions at all. Instead of updating the VCS repositories one-by-one, it just goes straight to `:: nothing to do... everything is up to date`.  I also read through `--optimistic_versioning` and `--ignore_versioning`, but didn't exactly understand these options.

How would you recommend updating development packages?

## Content of `aurman` config
Empty

## Read the README
Yes

## Running linux distribution
Arch Linux

> **Notice**: If you do not fill out one of the fields, without mentioning **why** you left it out, the issue will be closed without any comment.
You may also be banned from this repository without further warning.

I skipped a few fields, because I'm presuming this issue is from my stupidity, rather than a bug. Please don't ban me…!

**Comment by protist** [comment]
Thank you for the reply. Yes, that seems to work now. The packages are checked one by one.
```
$ aurman -Syu --devel --devel_skip_deps
:: Synchronising package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
:: Starting full system upgrade...
 there is nothing to do

~~ initializing aurman...
~~ looking for new pkgbuilds of devel packages and fetching them...
==> Making package: backintime-git 1.1.12.r443.g012b0783-1 (Sat 18 Aug 2018 21:37:21 AEST)
==> WARNING: Skipping dependency checks.
==> Retrieving sources...
  -> Updating backintime git repo...
Fetching origin
==> Validating source files with md5sums...
    backintime ... Skipped
==> Extracting sources...
  -> Creating working copy of backintime git repo...
Cloning into 'backintime'...
done.
==> Starting pkgver()...
==> Sources are ready.
==> Cleaning up...
```
And so on for all my development packages.

I wasn't clear in my original post, but I think I misunderstood what the command should have been. I replaced `--devel` with `--devel_skip_deps`, i.e. I ran the following:
```
$ aurman -Syu --devel_skip_deps --needed --noedit --noconfirm
```
So if I understand correctly, `--devel_skip_deps` is the correct option to use, but I still need to explicitly use `--devel` as well? Thanks again.

**Comment by polygamma** [comment]
It is definitely not the "correct option" to choose, but one of the possibilities. If you want to use `--devel_skip_deps`, you still need to also use `--devel`, yes, but that does not necessarily mean, that this is the option you "really" want to use. It is important, that you understand what all of this is about, I suggest to read issue 245

**Comment by polygamma** [comment]
Addition: Especially https://github.com/polygamma/aurman/issues/245#issuecomment-412674873

**Comment by polygamma** [comment]
tl;dr - the sane way is to actually fulfill the dependencies, because makepkg expects all dependencies to be available, even if "only" running prepare and pkgver.

**Comment by protist** [comment]
Thanks for that link. I think that all makes sense to me. Thank you for the detailed reply.
